Marlin/src/module/stepper.cpp

```diff
@@ -2094,7 +2094,8 @@ hal_timer_t Stepper::calc_timer_interval(uint32_t step_rate) {
 
   #ifdef CPU_32_BIT
 
-    return uint32_t(STEPPER_TIMER_RATE) / step_rate; // A fast processor can just do integer division
+    // A fast processor can just do integer division
+    return step_rate ? uint32_t(STEPPER_TIMER_RATE) / step_rate : HAL_TIMER_TYPE_MAX;
```

Comment:
i.e., Always returns something <= `STEPPER_TIMER_RATE`. If we had a float incoming, it would be possible to get values > `STEPPER_TIMER_RATE` for `0 < step_rate < 1`. Do we consider an incoming `step_rate` of 0 to be a value between 0 and 1? If so, the return value could be anything > `STEPPER_TIMER_RATE` and <= `HAL_TIMER_TYPE_MAX`.

Thus, it seems like `calc_timer_interval(current_block->some_rate) << current_block->la_scaling` might be more appropriately nuanced.

Reply:
If `step_rate < 1` we have a problem anyway. That means 1 whole second or more until the next ISR call. During acceleration, the speed would have changed immensely during that time. The stepper ISR code is built around the assumption that step rate is high enough that speed change between steps is reasonably small. The planner's `MINIMAL_STEP_RATE` should mean step rates never get below 120.

However for LA `HAL_TIMER_TYPE_MAX` is more than just "a long time from now" it is also a special value that says "keep checking whether la_interval has changed".